contrib_bots/lib/github_detail.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import print_function
+
+import re
+import os
+import sys
+import logging
+import six.moves.configparser
+
+import requests
+
+class GithubHandler(object):
+    '''
+    This bot provides details on github issues and pull requests when they're
+    referenced in the chat.
+    '''
+
+    CONFIG_PATH = os.path.expanduser('~/.contrib_bots/github_detail.ini')
+    GITHUB_ISSUE_URL_TEMPLATE = 'https://api.github.com/repos/{owner}/{repo}/issues/{id}'
+    GITHUB_PULL_URL_TEMPLATE = 'https://api.github.com/repos/{owner}/{reepo}/pulls/{id}'
+    HANDLE_MESSAGE_REGEX = re.compile("([\w-]+)?\/?([\w-]+)?#(\d+)")
+    MAX_LENGTH_OF_MESSAGE = 200
+
+    def __init__(self):
+        config = six.moves.configparser.ConfigParser()
+        with open(self.CONFIG_PATH) as config_file:
+            config.readfp(config_file)
+        if config.get('Github', 'owner'):
+            self.owner = config.get('Github', 'owner')
+        else:
+            # Allowing undefined default repos would require multiple triage_message regexs.
+            # It's simpler to require them to be defined.
+            sys.exit('Default owner not defined')
+
+        if config.get('Github', 'repo'):
+            self.repo = config.get('Github', 'repo')
+        else:
+            sys.exit('Default repo not defined')
+
+    def usage(self):
+        # type: () -> None
+        return ("This plugin displays details on github issues and pull requests. "
+                "To refrence an issue or pr usename mention the bot then anytime "
+                "in the message it's id, for example:\n"
+                "@**Github detail** #3212 zulip/#3212 zulip/zulip#3212\n"
+                "The default owner is {} and the default repo is {}.".format(self.owner, self.repo))
+
+    def triage_message(self, message, client):
+        # type: () -> bool
+        # Check the message contains a username mention, an issue idi
+        # or 'help', and that we're not replying to another bot.
+        regex = "(?:@(?:\*\*){}).+?(?:#\d+)|(?:help)".format(re.escape(client.full_name))
+        return re.search(regex, message['content']) and not message['sender_email'].endswith('-bot@zulip.com')
+
+    def format_message(self, details):
+        # type: (Dict[Text, Union[Text, int, bool]]) -> Text
+        number = details['number']
```

Comment:
Getting this when calling with the example provided (`@**Github detail** #3212 zulip/#3212 zulip/zulip#3212`) or any other input:
```
Traceback (most recent call last):
  File "run.py", line 124, in <module>
    run()
  File "run.py", line 120, in run
    quiet=options.quiet
  File "run.py", line 80, in run_message_handler_for_bot
    client.call_on_each_message(handle_message)
  File "../api/zulip/__init__.py", line 471, in call_on_each_message
    self.call_on_each_event(event_callback, ['message'])
  File "../api/zulip/__init__.py", line 463, in call_on_each_event
    callback(event)
  File "../api/zulip/__init__.py", line 470, in event_callback
    callback(event['message'])
  File "run.py", line 76, in handle_message
    state_handler=state_handler
  File "/Users/alicja/Zulip/zulip/contrib_bots/lib/github_detail.py", line 124, in handle_message
    bot_messages.append(self.format_message(details))
  File "/Users/alicja/Zulip/zulip/contrib_bots/lib/github_detail.py", line 56, in format_message
    number = details['number']
TypeError: 'NoneType' object has no attribute '__getitem__'
```